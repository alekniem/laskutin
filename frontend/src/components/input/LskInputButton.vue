<template>
    <v-btn :color="color" :disabled="disabled" :id="cid" depressed small>
        {{ buttonText | trans }}
    </v-btn>
</template>

<script>
export default {
    name: "LskInputButton",

    props: {
        buttonText: { type: String, required: true },
        cid: { type: String, required: true },
        color: { type: String, required: false, default: "primary" },
        disabled: { type: Boolean, required: false, default: false },
    },
};
</script>
