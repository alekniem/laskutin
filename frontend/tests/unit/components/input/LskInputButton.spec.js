import LskInputButton from "@/components/input/LskInputButton";
import { createWrapper } from "UT/lsk_test_helper";

describe("LskInputButton", () => {
    it("should match snapshot", () => {
        const wrapper = createWrapper(LskInputButton, {
            propsData: {
                buttonText: "TEST BUTTON TEXT",
                cid: "TEST_CID",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("default values");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskInputButton, {
            propsData: {
                buttonText: "TEST BUTTON TEXT",
                color: "purple",
                cid: "TEST_CID",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("purple color");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskInputButton, {
            propsData: {
                buttonText: "TEST BUTTON TEXT",
                color: "purple",
                disabled: true,
                cid: "TEST_CID",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("disabled");
    });
});
