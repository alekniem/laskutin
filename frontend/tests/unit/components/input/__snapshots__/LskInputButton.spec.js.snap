// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LskInputButton should match snapshot: default values 1`] = `
<button type="button"
        class="v-btn v-btn--has-bg theme--light v-size--small primary"
        id="TEST_CID"
>
  <span class="v-btn__content">
    TEST BUTTON TEXT
  </span>
</button>
`;

exports[`LskInputButton should match snapshot: disabled 1`] = `
<button type="button"
        disabled="disabled"
        class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--small"
        id="TEST_CID"
>
  <span class="v-btn__content">
    TEST BUTTON TEXT
  </span>
</button>
`;

exports[`LskInputButton should match snapshot: purple color 1`] = `
<button type="button"
        class="v-btn v-btn--has-bg theme--light v-size--small purple"
        id="TEST_CID"
>
  <span class="v-btn__content">
    TEST BUTTON TEXT
  </span>
</button>
`;
